json.go

```diff
@@ -90,16 +90,29 @@ func ConcatJSON(blobs ...[]byte) []byte {
 	if len(blobs) == 0 {
 		return nil
 	}
-	if len(blobs) == 1 {
+
+	last := len(blobs) - 1
+	for bytes.Equal(blobs[last], []byte("null")) || blobs[last] == nil {
```

Comment:
can you move []byte("null") to a constant / package level var, or replace it with: 

```go
lb := blobs[last]
for len(lb) == 0 || (len(lb) >= 4 && lb[0] == 'n' && lb[1] == 'u' && lb[2] == 'l' && lb[3] == 'l')
```

consider moving the nil check as first check so that it fails faster before trying to read.

Reply:
OK for moving the nil before. Good.
OK for a global var: this would avoid allocation

Regarding the scan optimization proposal, I am not sure this would be much better than the optimized inline assembly-language used by bytes to implement bytes.Equal()